.github/workflows/ci.yml

```diff
@@ -72,11 +87,13 @@ jobs:
   clippy:
     runs-on: ubuntu-latest
     timeout-minutes: 30
+    permissions:
+      actions: write
     steps:
       - uses: actions/checkout@v5
-      - uses: dtolnay/rust-toolchain@stable
         with:
-          components: clippy
+          persist-credentials: false
+      - uses: dtolnay/rust-toolchain@clippy
```

Comment:
note: this now uses `@clippy` in line with other Foundry crates which is `nightly + clippy` shorthand